File: imageSmoother.swift
class Solution {
    func imageSmoother(_ img: [[Int]]) -> [[Int]] {
        let m = img.count, n = img[0].count
        var res = Array(repeating: Array(repeating: 0, count: n), count: m)
        
        for i in 0..<m {
            for j in 0..<n {
                var sum = 0, count = 0
                for dx in -1...1 {
                    for dy in -1...1 {
                        let ni = i + dx, nj = j + dy
                        if ni >= 0 && ni < m && nj >= 0 && nj < n {
                            sum += img[ni][nj]
                            count += 1
                        }
                    }
                }
                res[i][j] = sum / count
            }
        }
        
        return res
    }
}

File: imageSmoother.go
func imageSmoother(img [][]int) [][]int {
    m, n := len(img), len(img[0])
    res := make([][]int, m)
    
    for i := range res {
        res[i] = make([]int, n)
        for j := range res[i] {
            sum, count := 0, 0
            for dx := -1; dx <= 1; dx++ {
                for dy := -1; dy <= 1; dy++ {
                    ni, nj := i + dx, j + dy
                    if ni >= 0 && ni < m && nj >= 0 && nj < n {
                        sum += img[ni][nj]
                        count++
                    }
                }
            }
            res[i][j] = sum / count
        }
    }
    
    return res
}
